libfdt/fdt_addresses.c

```diff
@@ -73,7 +73,7 @@ int fdt_appendprop_addrrange(void *fdt, int parent, int nodeoffset,
 	/* check validity of address */
 	prop = data;
 	if (addr_cells == 1) {
-		if ((addr > UINT32_MAX) || ((UINT32_MAX + 1 - addr) < size))
+		if ((addr > UINT32_MAX) || ((UINT32_MAX + 1ULL - addr) < size))
```

Comment:
Nice catch.  However, a small detail, I'd prefer to see an explicit case of UINT32_MAX to (uint64_t), which should enforce the same promotion.  `1ULL` will probably have the same type as `uint64_t` and therefore as `addr` and `size`, but it's nice to be explicit.

Reply:
Alright.